File: Polymorphsim/Polymorphsim.php
<?php
class Produk {
    public $merek;
    public $stok;

    public function __construct($merek, $stok) {
        $this->merek = $merek;
        $this->stok = $stok;
    }
    public function cekStok() {
        return "Stok saat ini: " . $this->stok;
    }
}

class Televisi extends Produk {
    public $jenis;
    public function __construct($merek, $stok, $jenis) {
        parent::__construct($merek, $stok);
        $this->jenis = $jenis;
    }
    public function cekStokTelevisi() {
        return "Jenis: " . $this->jenis . ", " . $this->cekStok() . "<br>";
    }
}

$produk01 = new Televisi("Samsung", 30, "LED");
echo $produk01->cekStokTelevisi();

#2
class Kendaraan {
    public $merk;
    public $tahunProduksi;

    public function __construct($merk, $tahunProduksi) {
        $this->merk = $merk;
        $this->tahunProduksi = $tahunProduksi;
    }
    public function infoKendaraan() {
        return "Merk: " . $this->merk . ", Tahun Produksi: " . $this->tahunProduksi;
    }
}

class Mobil extends Kendaraan {
    public $jenis;

    public function __construct($merk, $tahunProduksi, $jenis) {
        parent::__construct($merk, $tahunProduksi);
        $this->jenis = $jenis;
    }
    public function infoMobil() {
        return "Jenis: " . $this->jenis . ", " . $this->infoKendaraan() . "<br>";
    }
}

// Contoh penggunaan
$mobil1 = new Mobil("Toyota", 2020, "Sedan");
echo $mobil1->infoMobil();

#3
class Hewan {
    public $nama;
    public $jenis;

    public function __construct($nama, $jenis) {
        $this->nama = $nama;
        $this->jenis = $jenis;
    }
    public function infoHewan() {
        return "Nama: " . $this->nama . ", Jenis: " . $this->jenis;
    }
}

class Kucing extends Hewan {
    public $warnaBulu;

    public function __construct($nama, $jenis, $warnaBulu) {
        parent::__construct($nama, $jenis);
        $this->warnaBulu = $warnaBulu;
    }
    public function infoKucing() {
        return "Warna Bulu: " . $this->warnaBulu . ", " . $this->infoHewan();
    }
}

// Contoh penggunaan
$kucing1 = new Kucing("Whiskers", "Persia", "Coklat");
echo $kucing1->infoKucing();

?>

File: Polymorphsim/Polymorphsim.py
class Produk:
    def __init__(self, merek, stok):
        self.merek = merek
        self.stok = stok

    def cek_stok(self):
        return f"Stok saat ini: {self.stok}"

class Televisi(Produk):
    def __init__(self, merek, stok, jenis):
        super().__init__(merek, stok)
        self.jenis = jenis

    def cek_stok_televisi(self):
        return f"Jenis: {self.jenis}, {super().cek_stok()}"

# Contoh penggunaan
produk01 = Televisi("LG", 40, "LED")
print(produk01.cek_stok_televisi())

#2
class Kendaraan:
    def __init__(self, merk, tahun_produksi):
        self.merk = merk
        self.tahun_produksi = tahun_produksi

    def info_kendaraan(self):
        return f"Merk: {self.merk}, Tahun Produksi: {self.tahun_produksi}"

class Mobil(Kendaraan):
    def __init__(self, merk, tahun_produksi, jenis):
        super().__init__(merk, tahun_produksi)
        self.jenis = jenis

    def info_mobil(self):
        return f"Jenis: {self.jenis}, {super().info_kendaraan()}"

# Contoh penggunaan
mobil1 = Mobil("Nissan", 2021, "Hatchback")
print(mobil1.info_mobil())
    

#3
class Hewan:
    def __init__(self, nama, jenis):
        self.nama = nama
        self.jenis = jenis

    def info_hewan(self):
        return f"Nama: {self.nama}, Jenis: {self.jenis}"

class Kelinci(Hewan):
    def __init__(self, nama, jenis, warna_bulu):
        super().__init__(nama, jenis)
        self.warna_bulu = warna_bulu

    def info_kelinci(self):
        return f"Warna Bulu: {self.warna_bulu}, {super().info_hewan()}"

# Contoh penggunaan
kelinci1 = Kelinci("Fluffy", "Angora", "Putih");
print(kelinci1.info_kelinci())
